client/driver/mock_driver.go

```diff
@@ -0,0 +1,168 @@
+package driver
```

Comment:
I think we should add a build flag for `// +build nomad_test` which adds the mock driver to the BuiltinDrivers and gate this file